lib/compatibility.bzl

```diff
@@ -0,0 +1,130 @@
+"""Skylib module of convenience functions for `target_compatible_with`.
+
+Load the macros as follows in your `BUILD` files:
+```python
+load("@bazel_skylib//lib:compatibility.bzl", "compatibility")
+```
+
+See the [Platform docs](https://bazel.build/docs/platforms#skipping-incompatible-targets) for
+more information.
+"""
+
+load(":selects.bzl", "selects")
+load("//lib/compatibility:defs.bzl", "MAX_NUM_ALL_OF_SETTINGS")
+
+def _none_of(*settings):
+    """Create a `select()` for `target_compatible_with` which matches none of the given settings.
+
+    Any of the settings will resolve to an incompatible `constraint_value` for the
+    purpose of target skipping.
+
+    In other words, use this function to make target incompatible if any of the settings are true.
+
+    ```python
+    cc_binary(
+        name = "bin",
+        srcs = ["bin.cc"],
+        # This target cannot be built for Linux or Mac, but can be built for
+        # everything else.
+        target_compatible_with = compatibility.none_of(
+            "@platforms//os:linux",
+            "@platforms//os:macos",
+        ),
+    )
+    ```
+
+    Args:
+      *settings: The `config_setting` or `constraint_value` targets.
+
+    Returns:
+      A native `select()` which maps any of the settings to the incompatible target.
+    """
+    return selects.with_or({
+        "//conditions:default": [],
+        tuple(settings): ["@bazel_skylib//lib/compatibility:none_of"],
```

Comment:
Nit: I would find this slight more "understandable" if you put "//conditions:default" at the end. If I understand the semantics of select correctly - it's evaluated top to bottom, with default always being last.

Reply:
That's fair. I will fix that in the next iteration.